flink-connectors/flink-hbase/src/main/java/org/apache/flink/addons/hbase/HBaseTableSchema.java

```
@@ -0,0 +1,135 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.addons.hbase;
+
+import org.apache.flink.api.common.typeinfo.BasicArrayTypeInfo;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.util.Preconditions;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.hadoop.hbase.util.Pair;
+
+import java.io.Serializable;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.ArrayList;
+import java.util.Date;
+
+/**
+ * Helps to specify an HBase Table's schema
+ */
+public class HBaseTableSchema implements Serializable {
+
+	// A Map with key as column family.
+	private final Map<String, List<Pair>> familyMap =
+		new HashMap<String, List<Pair>>();
+
+	// Allowed types. This may change.
+	// TODO : Check if the Date type should be the one in java.util or the one in java.sql
+	private static Class[] CLASS_TYPES = {
+		Integer.class, Short.class, Float.class, Long.class, String.class, Byte.class, Boolean.class, Double.class, BigInteger.class, BigDecimal.class, Date.class
+	};
+	private static byte[] EMPTY_BYTE_ARRAY = new byte[0];
+	public void addColumns(String family, String qualifier, TypeInformation<?> type) {
+		Preconditions.checkNotNull(family, "family name");
+		Preconditions.checkNotNull(family, "qualifier name");
+		Preconditions.checkNotNull(type, "type name");
+		List<Pair> list = this.familyMap.get(family);
+		if (list == null) {
+			list = new ArrayList<Pair>();
+		}
+		boolean found = false;
+		for(Class classType : CLASS_TYPES) {
+			if(classType == type.getTypeClass()) {
+				found = true;
+				break;
+			}
+		}
+		if(!found) {
+			// by default it will be byte[] type only
+			type = BasicArrayTypeInfo.BYTE_ARRAY_TYPE_INFO;
```

Ok I will add byte[].class into the CLASS_TYPES. And anything other than the ones in CLASS_TYPES we can throw an exception.